src/json-symtab-language/json_symtab_language.cpp

```diff
@@ -39,6 +41,65 @@ bool json_symtab_languaget::typecheck(
   }
 }
 
+void json_symtab_languaget::follow_symbol_types(
+  irept &irep,
+  const namespacet &ns)
+{
+  if(irep.id() == ID_symbol)
```

Comment:
Can't this whole if block be replaced with a single call to ns.follow_symbol(irep) ?

Reply:
The `follow_symbol()` method follows both type symbols and non-type symbols, but we only want to replace type symbols. So we first do the check `if(symbol.is_type)` and only do the follow in this case.

Reply:
Ah, right - got it... A couple of observations:

1) This should probably be renamed to `follow_type_symbols` to match the naming of `follow_symbols`
2) Should either this `follow_type_symbols` function be moved into `namespace.cpp` as a utility function there, or should ns.follow_symbol() grow an extra argument (with a default value) that specifies whether to follow non-type symbols or not, and then just use that where we need it instead?

Reply:
Ah yes. I think I'll move the new `follow_type_symbols` function to `namespace.cpp`.